community/kernel/src/main/java/org/neo4j/kernel/api/txtracking/TransactionIdTracker.java

```diff
@@ -78,23 +74,22 @@ public void awaitUpToDate( long oldestAcceptableTxId, Duration timeout ) throws
             return;
         }
 
-        if ( !tryAwaitEx( () -> isReady( oldestAcceptableTxId ), timeout.toMillis(), TimeUnit.MILLISECONDS,
-                POLL_INTERVAL, POLL_UNIT ) )
-        {
-            throw new TransactionFailureException( Status.Transaction.InstanceStateChanged,
-                    "Database not up to the requested version: %d. Latest database version is %d", oldestAcceptableTxId,
-                    transactionIdStore().getLastClosedTransactionId() );
-        }
-    }
-
-    private boolean isReady( long oldestAcceptableTxId ) throws TransactionFailureException
-    {
         if ( !availabilityGuard.isAvailable() )
         {
             throw new TransactionFailureException( Status.General.DatabaseUnavailable,
                     "Database had become unavailable while waiting for requested version %d.", oldestAcceptableTxId );
         }
-        return oldestAcceptableTxId <= transactionIdStore().getLastClosedTransactionId();
+
+        try
+        {
+            transactionIdStore().awaitClosedTransactionId( oldestAcceptableTxId, timeout.toMillis() );
+        }
+        catch ( InterruptedException | TimeoutException e )
```

Comment:
I think interruption status of the thread should be restored here, if it was actually interrupted